src/components/motion/MotionController.cpp

```diff
@@ -22,11 +22,10 @@ void MotionController::Update(int16_t x, int16_t y, int16_t z, uint32_t nbSteps)
   lastZ = this->z;
   this->z = z;
 
-  int32_t deltaSteps = nbSteps - this->nbSteps;
-  this->nbSteps = nbSteps;
-  if (deltaSteps > 0) {
-    currentTripSteps += deltaSteps;
+  if (nbSteps > this->nbSteps) {
+    currentTripSteps += nbSteps - this->nbSteps;
   }
+  this->nbSteps = nbSteps;
 }
```

Comment:
The previous code could handle nbSteps overflowing... in case someone walks 2^32 steps in a day... Regardless maybe we don't need to change this.

Reply:
I don't see how my code is any worse at handling overflows than the old code. Seeing as nbSteps is stored as a `uint32_t`, overflows just reset it to 0, and so I believe the code does the exact same.

Anyway, I mostly wanted to change this because the conversion from unsigned to signed didn't sit entirely right to me, and this new code is more obvious to me. If the old code makes more sense to you, I'd be happy to revert it though.

Reply:
If nbSteps overflows, it's not larger than the previous steps, so the trip steps isn't updated. Using the deltaSteps variable, subtracting 4294967276 from 20 results in 40, so it doesn't miss the trip steps.

Reply:
Oh, yeah you're right. Oops. Anyway, seeing as that is an impossibility in reality (unless the accelerometer is returning bogus values, in which case it doesn't matter anyway), I think what matters more is how nicely the code reads. What do you think?

Reply:
I think the old code is the most readable after moving this->nbSteps away from the middle.